File: backend/Shared/EventPlanR.Domain/Common/Address.cs
namespace EventPlanr.Domain.Common;

public class Address
{
    public string Country { get; set; } = null!;
    public string ZipCode { get; set; } = null!;
    public string City { get; set; } = null!;
    public string AddressLine { get; set; } = null!;
}

File: frontend/lib/domain/models/common/address.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'address.freezed.dart';
part 'address.g.dart';

@freezed
class Address with _$Address {
  const factory Address({
    required String country,
    required String zipCode,
    required String city,
    required String addressLine,
  }) = _Address;

  factory Address.fromJson(Map<String, Object?> json) =>
      _$AddressFromJson(json);
}
